File: SoftUni Lessons/Python Development/Python Basics April 2019/Lessons and Problems/11.1 - For Loop More Exercises/04. Grades.py
students = int(input())

n1 = 0
n2 = 0
n3 = 0
n4 = 0
total_average_score = 0


for i in range(1, students + 1):
    score = float(input())
    total_average_score += score
    if 2.00 <= score <= 2.99:
        n1 += 1
    elif 3.00 <= score <= 3.99:
        n2 += 1
    elif 4.00 <= score <= 4.99:
        n3 += 1
    elif score >= 5:
        n4 += 1

print(f"Top students: {(n4 / students * 100):.2f}%")
print(f"Between 4.00 and 4.99: {(n3 / students * 100):.2f}%")
print(f"Between 3.00 and 3.99: {(n2 / students * 100):.2f}%")
print(f"Fail: {(n1 / students * 100):.2f}%")
print(f"Average: {(total_average_score / students):.2f}")

File: SoftUni Lessons/JavaScript Development/JavaScript Basics March 2020/12. УПРАЖНЕНИЕ ЦИКЛИ/For-Loop - More Exercises/04. Grades.js
function grades(input) {
    // Mask
    let students = Number(input.shift());
    
    let n1 = 0;
    let n2 = 0;
    let n3 = 0;
    let n4 = 0;
    let total = 0;

    for (let student = 1; student <= students; student += 1) {
        score = Number(input.shift());
        total += score

        if (score >= 2 && score <= 2.99) {
            n1 += 1;
        } else if (score >= 3 && score <= 3.99) {
            n2 += 1;
        } else if (score >= 4 && score <= 4.99) {
            n3 += 1;
        } else {
            n4 += 1
        }
    }

    console.log(`Top students: ${(n4 / students * 100).toFixed(2)}%`)
    console.log(`Between 4.00 and 4.99: ${(n3 / students * 100).toFixed(2)}%`)
    console.log(`Between 3.00 and 3.99: ${(n2 / students * 100).toFixed(2)}%`)
    console.log(`Fail: ${(n1 / students * 100).toFixed(2)}%`)
    console.log(`Average: ${(total / students).toFixed(2)}`)
}

grades([10, 3.00, 2.99, 5.68, 3.01, 4, 4, 6.00, 4.50, 2.44, 5]);  // Should return:
// Top students: 30.00%
// Between 4.00 and 4.99: 30.00%
// Between 3.00 and 3.99: 20.00%
// Fail: 20.00%
// Average: 4.06

grades([6, 2, 3, 4, 5, 6, 2.2]);  // Should return:
// Top students: 33.33%
// Between 4.00 and 4.99: 16.67%
// Between 3.00 and 3.99: 16.67%
// Fail: 33.33%
// Average: 3.70
